### DISCUSSION TITLE
When to you hooks.client.ts

### DISCUSSION THREAD

**Discussion by Vrajs16** [sveltejs/kit#9152] [General]
So I understand that hooks.server.ts runs on every request but when i use hooks.client.ts, it only runs once when I refresh the page. If someone could explain when I would use this file, that would be greatly appreciated.

**Comment by teemingc** [comment]
That depends on what you're exporting from your hooks file. So far, the only export for client hooks is to handle errors. Exporting handle from your client hook will have no effect.

https://kit.svelte.dev/docs/hooks#shared-hooks

**Comment by jonshipman** [comment]
hooks.client.ts looks to run every +layout.svelte change. (sveltekit 2)

**Comment by david-plugge** [comment]
hooks.client.ts runs initially when the client gets loaded and before the client actually takes over rendering (before hydration). The reason this file exists is because of client hooks like handleError and I suggest you only use it for that.

It never runs on the server, only on the client for the initial load.
